### PR TITLE
Add cache system

### PR DESCRIPTION
**Pull request by bidoubiwa**
TODO:
 - [x] tests

Feature: 
- Use cache on pagination changes
- Same requests uses same cache

fixes: #520

### CODE REVIEW

File: tests/search-resolver.tests.ts [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    await searchClient.search<Movies>([searchParameters1])
+    await searchClient.search<Movies>([searchParameters2])
+
+    expect(mockedMeilisearch).toHaveBeenCalledWith({
+      host: 'http://localhost:7700',
+      apiKey: '',
+    })
+    expect(mockedSearch).toHaveBeenCalledTimes(2)
+  })
+  test('Test two same and one different search parameter', async () => {
```
Comment (alallema) [inline tests/search-resolver.tests.ts:79]:
```suggestion
  })
  
  test('Test two identical and one different search parameters, async () => {
```

File: tests/search-resolver.tests.ts [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+    await searchClient.search<Movies>([searchParameters1])
+
+    expect(mockedMeilisearch).toHaveBeenCalledWith({
+      host: 'http://localhost:7700',
+      apiKey: '',
+    })
+    expect(mockedSearch).toHaveBeenCalledTimes(2)
+  })
+
+  test('Test two same and two different search parameter', async () => {
```
Comment (alallema) [inline tests/search-resolver.tests.ts:105]:
```suggestion
  test('Test two identical and two different search parameter', async () => {
```

**Review by alallema — COMMENTED** [review]
I remove some `console.log()` but maybe they were there on purpose

File: tests/search-resolver.tests.ts (hunk 1 above)
Comment (bidoubiwa) [reply tests/search-resolver.tests.ts:79]: You forgot a closing quote

File: tests/search-resolver.tests.ts (hunk 2 above)
Comment (bidoubiwa) [reply tests/search-resolver.tests.ts:105]: I think this means the same no?

File: tests/search-resolver.tests.ts (hunk 1 above)
Comment (bidoubiwa) [reply tests/search-resolver.tests.ts:79]:
```suggestion
 })
  
  test('Test two identical and one different search parameters', async () => {
```

**Review by alallema — APPROVED** [review]
LGTM! 🎉

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**